website/source/docs/internals/providers/new-data-source.html.md

```diff
@@ -0,0 +1,90 @@
+---
+layout: "docs"
+page_title: "Creating Data Sources"
+sidebar_current: "docs-internals-provider-guide-new-data-source"
+description: |-
+  How to get started adding a data source to an existing provider.
+---
+
+# Creating Data Sources
+
+Once we have a Provider, we can add some Data Sources to that provider, which
+will help it pull information out of your infrastructure to be used in your
+Resources.  To do this, you need to write some code in the package your
+Provider is defined in.
+
+In the built-in plugins for Terraform that are shipped as part of the codebase,
+we have a file naming guideline: we use
+`data_source_{api}_{data_source_name}.go` as the template to name our files,
+and tend to stick to  single Data Source per file. For example, AWS has
+`data_source_aws_acm_certificate.go`, because it reads certificates from AWS’
+ACM service. Some providers only have a single API, so we use the provider name
+in that case. For example, Docker has `data_source_docker_registry_image.go`.
+If you want to contribute your code back to the Terraform repo, it makes things
+easier if you follow this convention.
+
+Inside this file, you’re going to define a function that takes no arguments and
```

Comment:
I don't have code samples at the moment, and do not believe I'd be able to come up with samples of the appropriate quality to meet the business timeline for publishing this. (~sometime next week, earlier the better).